sdks/python/apache_beam/runners/worker/opcounters_test.py

```diff
@@ -41,6 +47,55 @@ def __init__(self):
     pass
 
 
+class TransformIoCounterTest(unittest.TestCase):
+
+  def setUp(self):
+    if not statesampler.FAST_SAMPLER:
+      raise SkipTest('State sampler not compiled.')
+    super(TransformIoCounterTest, self).setUp()
+
+  def test_basic_counters(self):
+    counter_factory = CounterFactory()
+    sampler = statesampler.StateSampler('stage1', counter_factory)
+    sampler.start()
+
+    with sampler.scoped_state('step1', 'stateA'):
+      counter = opcounters.SideInputReadCounter(counter_factory, sampler,
+                                                declaring_step='step1',
+                                                input_index=1)
+    with sampler.scoped_state('step2', 'stateB'):
+      with counter:
+        counter.add_bytes_read(10)
+        time.sleep(0.1)  # An arbitrary short interval.
```

Comment:
<!--thread_id:cc_163728407_t; commit:b68f9f26428de8ed2f34c0919ee91a1662402d86; resolved:1-->
<!--section:context-quote-->
> **aaltay** wrote:
> What are we testing by this sleep?

<!--section:body-->
The idea is to check that time was sampled under the time counter, but I guess this can be removed from now, while we experiment with the time counter and performance..